### DISCUSSION TITLE
Python newbie - enrollment app not quite doing what I want

### DISCUSSION THREAD

**Discussion by RAA-git** [community/community#140550] [Programming Help]
### Body

Creating a practice enrollment python program that allows a user to “enroll” in an IT course. More information in the read me file of the repository.

My issue is everything is ok except for the selection in the Diploma, selecting either course in the prerequisite will not bring up the option for selecting the Diploma to enroll in.

https://github.com/RAA-git/ICTPRG-task2.git

The user must first choose the courses they have already completed. The user then may only choose a course where they meet the “Prerequisite”.
After the user selects the appropriate course, an enrolment confirmation message must be shown to the user. (The application does NOT need to enroll a user!)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by prblsing** [comment]
Hey @RAA-git,
It looks like you’re on the right track with your enrollment app! 👏 However, I couldn't find the complete code that you’re working on in the repository link you shared. To help you better, could you upload your code or share the specific part where the issue occurs?
This will allow us to review it and suggest the exact changes needed to get the prerequisite validation working as expected. If you prefer not to share the entire code, even a snippet or log showing the problem will be useful.
Once you share that, I can help you address the issue of selecting the right Diploma based on the prerequisite.
Cheers! 😊

**Comment by R0b0t-Maker** [comment]
Based on your issue, it sounds like there might be a logical problem with how the prerequisite selection and Diploma options are being connected. Here are a few suggestions to troubleshoot this:

1. Ensure that the logic used to determine the prerequisite match is correct. Verify that each course properly maps to a Diploma.
2. After selecting the prerequisite course, check that the Diploma options are being dynamically updated based on the selection. This could be an issue with how events are triggered in your code.
3. Look for any conditional checks that may be inadvertently preventing the Diploma options from being displayed. Make sure the flow progresses properly after selecting prerequisites.

If you are using a graphical interface, ensure that the diploma options are visually updated after prerequisites are selected. This could involve refreshing the UI elements.

If you could provide the specific snippet of code handling the prerequisite checks and Diploma options, I can help you debug further.
